.github/workflows/tests.yml

```diff
@@ -0,0 +1,131 @@
+name: Run InVEST Tests
+
+on: [push, pull_request, check_run]
+
+jobs:
+    # This job will check for obvious syntax errors.
+    # The other jobs in this file depend on this one and won't show up in the
+    # jobs list on actions until this one completes successfully.
+    # Best to run in linux ... windows won't honor nonzero exit statuses
+    # by default.
+    check-syntax-errors:
+        name: "Check for syntax errors"
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v1
+
+            - name: Set up python 3.7
+              uses: actions/setup-python@v1
+              with:
+                  python-version: 3.7
+
+            - name: Set up environment
+              run: python -m pip install --upgrade pip setuptools setuptools_scm flake8
+
+            - name: Lint with flake8
+              run: |
+                # stop the build if there are Python syntax errors or undefined names
+                python -m flake8 src --count --select=E9,F63,F7,F82 --show-source --statistics
+                # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+                python -m flake8 src --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+
+    run-model-tests:
+        name: "Run model tests"
+        runs-on: ${{ matrix.os }}
+        needs: check-syntax-errors
+        strategy:
+            fail-fast: false
+            max-parallel: 4
+            matrix:
+                python-version: [3.6, 3.7]
+                python-architecture: [x86, x64]
+                os: [windows-latest]
+
+        steps:
+            - uses: actions/checkout@v1
+
+              # Taken from https://github.com/actions/cache/blob/master/examples.md#python---pip
+            - name: Restore pip cache
+              uses: actions/cache@v1
+              with:
+                  path: ~\AppData\Local\pip\Cache
+                  key: ${{ runner.os }}-py${{ matrix.python-version }}-${{ matrix.python-architecture }}-pip-${{ hashFiles('**/requirements*.txt') }}-model-tests
+
+            # NOTE: It takes twice as long to save the sample data cache
+            # as it does to do a fresh clone (almost 5 minutes vs. 2.5 minutes)
```

Comment:
Yep, you're absolutely right about the redundancy here.  I went with option 1 in rev b998e1d and moved the `make validate_sampledata` step to its own build job outside of the build matrix.  With this task able to happen in parallel with the other jobs, the total build time dropped 22% from about 14 minutes to about 11 minutes. 🎉 

It looks like we had the sampledata validation tests in this repo because they do a `from natcap.invest import datastack`.  We could, of course, move these validation tests over to the other repo, but I think it makes sense to keep them here and in sync with any changes we're making to validation itself.

Reply:
Cool @phargogh , I'm happy with this!

FYI I think marking the conversation as resolved hid it from the Conversation page, which is not so convenient.  Overall I wish github PRs had fewer bells & whistles like this. Maybe I'll start liking it more eventually.

Reply:
Ah, sorry @davemfish , that was my mistake marking the conversation as resolved.  Sorry about that!